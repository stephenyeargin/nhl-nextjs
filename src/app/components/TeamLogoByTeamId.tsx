'use client';

import React from 'react';
import TeamLogo from './TeamLogo';

type BasicTeam = {
  id?: number | string;
  abbrev: string;
  logo?: string;
};

interface TeamLogoByTeamIdProps {
  teamId?: number | string;
  homeTeam: BasicTeam;
  awayTeam: BasicTeam;
  size?: number; // Tailwind size token like 10, 12, 16
  theme?: 'light' | 'dark' | 'auto' | undefined;
  className?: string;
}

const TeamLogoByTeamId: React.FC<TeamLogoByTeamIdProps> = ({
  teamId,
  homeTeam,
  awayTeam,
  size,
  theme,
  className,
}) => {
  let computedClassName = className || 'h-10 w-10';
  if (size) {
    computedClassName = `h-${size} w-${size}`;
  }

  if (teamId === awayTeam.id) {
    return (
      <TeamLogo
        src={awayTeam.logo}
        alt={awayTeam.abbrev}
        className={computedClassName}
        colorMode={theme as any}
      />
    );
  }
  if (teamId === homeTeam.id) {
    return (
      <TeamLogo
        src={homeTeam.logo}
        alt={homeTeam.abbrev}
        className={computedClassName}
        colorMode={theme as any}
      />
    );
  }

  return null;
};

export default TeamLogoByTeamId;
