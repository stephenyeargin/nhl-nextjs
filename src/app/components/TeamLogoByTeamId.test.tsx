import React from 'react';
import { render, screen } from '@testing-library/react';
import TeamLogoByTeamId from './TeamLogoByTeamId';

// Mock TeamLogo to a simple div that echoes props
jest.mock('./TeamLogo', () => {
  const Mock = ({ alt, className }: any) => (
    <div data-testid="team-logo" data-alt={alt} className={className} />
  );
  (Mock as any).displayName = 'TeamLogoMock';

  return Mock;
});

const homeTeam = { id: 1, abbrev: 'NSH', logo: '/nsh.png' };
const awayTeam = { id: 2, abbrev: 'CBJ', logo: '/cbj.png' };

describe('TeamLogoByTeamId', () => {
  it('renders home team logo when teamId matches home', () => {
    const { asFragment } = render(
      <TeamLogoByTeamId
        teamId={1}
        homeTeam={homeTeam}
        awayTeam={awayTeam}
        size={16}
        theme="light"
      />
    );
    const logo = screen.getByTestId('team-logo');
    expect(logo).toHaveAttribute('data-alt', 'NSH');
    expect(logo).toHaveClass('h-16');
    expect(logo).toHaveClass('w-16');
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders away team logo when teamId matches away', () => {
    const { asFragment } = render(
      <TeamLogoByTeamId teamId={2} homeTeam={homeTeam} awayTeam={awayTeam} />
    );
    const logo = screen.getByTestId('team-logo');
    expect(logo).toHaveAttribute('data-alt', 'CBJ');
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders nothing when teamId is unknown', () => {
    const { container, asFragment } = render(
      <TeamLogoByTeamId teamId={999} homeTeam={homeTeam} awayTeam={awayTeam} />
    );
    expect(container).toBeEmptyDOMElement();
    expect(asFragment()).toMatchSnapshot();
  });
});
