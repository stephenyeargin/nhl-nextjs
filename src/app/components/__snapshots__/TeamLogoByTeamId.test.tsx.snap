// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamLogoByTeamId renders away team logo when teamId matches away 1`] = `
<DocumentFragment>
  <div
    class="h-10 w-10"
    data-alt="CBJ"
    data-testid="team-logo"
  />
</DocumentFragment>
`;

exports[`TeamLogoByTeamId renders home team logo when teamId matches home 1`] = `
<DocumentFragment>
  <div
    class="h-16 w-16"
    data-alt="NSH"
    data-testid="team-logo"
  />
</DocumentFragment>
`;

exports[`TeamLogoByTeamId renders nothing when teamId is unknown 1`] = `<DocumentFragment />`;
